openwisp_controller/config/base/config.py

```diff
@@ -590,12 +605,13 @@ def get_vpn_context(self):
             if vpn.subnet:
                 if vpnclient.ip:
                     context[vpn_context_keys['ip_address']] = vpnclient.ip.ip_address
-            if 'vni' in vpn_context_keys and (
-                vpnclient.vni or vpnclient.vpn._vxlan_vni
-            ):
+            if vpnclient.zt_identity_secret:
+                context[vpn_context_keys['member_id']] = vpnclient.member_id
                 context[
-                    vpn_context_keys['vni']
-                ] = f'{vpnclient.vni or vpnclient.vpn._vxlan_vni}'
+                    vpn_context_keys['zt_identity_secret']
+                ] = vpnclient.zt_identity_secret
+            if 'vni' in vpn_context_keys and vpnclient.vni:
+                context[vpn_context_keys['vni']] = f'{vpnclient.vni}'
```

Comment:
There are some inadvertent changes here. 

```suggestion
            if 'vni' in vpn_context_keys and (
                vpnclient.vni or vpnclient.vpn._vxlan_vni
            ):
                context[
                    vpn_context_keys['vni']
                ] = f'{vpnclient.vni or vpnclient.vpn._vxlan_vni}'
```